src/test/java/ch/wisv/areafiftylan/products/service/TicketServiceTest.java

```diff
@@ -0,0 +1,664 @@
+    }
+
+    @Test
+    public void getNumberSoldOfTypeTypeNull() {
+        assertEquals(Integer.valueOf(0), ticketService.getNumberSoldOfType(null));
+    }
+
+    @Test
+    public void findValidTicketsByOwnerUsername() {
+        //TODO I can't get this method to do my bidding...
+        /*
+        User user = persistUser();
+        Ticket ticket = ticketService.requestTicketOfType(
+        user, TEST_TICKET, Arrays.asList(CH_MEMBER_OPTION, PICKUP_SERVICE_OPTION));
+        testEntityManager.persist(ticket);
+        Collection<Ticket> tickets = ticketService.findValidTicketsByOwnerUsername(user.getUsername());
+        assertEquals(Collections.singleton(ticket), tickets);
+        */
+    }
+
+    @Test
+    public void getAllTicketsZero() {
+        Collection<Ticket> tickets = ticketService.getAllTickets();
+        assertEquals(0, tickets.size());
+    }
+
+    @Test
+    public void getAllTicketsOne() {
+        persistTicket();
+        Collection<Ticket> tickets = ticketService.getAllTickets();
+        assertEquals(1, tickets.size());
+    }
+
+    @Test
+    public void getAllTicketsTwo() {
+        persistTicket();
+        persistTicket();
+        Collection<Ticket> tickets = ticketService.getAllTickets();
+        assertEquals(2, tickets.size());
+    }
+
+    @Test
+    public void validateTicketFromInvalid() {
+        Ticket ticket = persistTicket();
+        ticket.setValid(false);
+        testEntityManager.persist(ticket);
+        assertEquals(false, ticket.isValid());
+        ticketService.validateTicket(ticket.getId());
+        assertEquals(true, ticket.isValid());
+    }
+
+    @Test
+    public void validateTicketAlreadyValid() {
+        Ticket ticket = persistTicket();
+        ticket.setValid(true);
+        testEntityManager.persist(ticket);
+        assertEquals(true, ticket.isValid());
+        ticketService.validateTicket(ticket.getId());
+        assertEquals(true, ticket.isValid());
+    }
+
+    @Test
+    public void validateTicketNotFound() {
+        Long id = 9999L;
+        thrown.expect(TicketNotFoundException.class);
+        thrown.expectMessage("Ticket not found");
+        ticketService.validateTicket(id);
+    }
+
+    @Test
+    public void validateTicketNull() {
+        thrown.expect(TicketNotFoundException.class);
+        thrown.expectMessage("Ticket not found");
+        ticketService.validateTicket(null);
+    }
+
+    @Test
+    public void requestTicketOfTypeStrings() {
+        User user = persistUser();
+        TicketType type = ticketTypeRepository.findByName(TEST_TICKET).get();
+        TicketOption pickupOption = ticketOptionRepository.findByName(PICKUP_SERVICE_OPTION).get();
+        TicketOption chMemberOption = ticketOptionRepository.findByName(CH_MEMBER_OPTION).get();
+        List<TicketOption> options = Arrays.asList(pickupOption, chMemberOption);
+        Set<TicketOption> optionSet = new HashSet<>(options);
```

Comment:
You can insert the previous `Arrays.asList` call in the `HashSet` constructor, removes one line and allows another newline for readability.